k2/csrc/fsa_utils.cu

```diff
@@ -1789,4 +1789,31 @@ Ragged<int32_t> ComposeArcMaps(Ragged<int32_t> &step1_arc_map,
   return Index(step1_arc_map, step2_arc_map, true);
 }
 
+void FixNumStates(FsaVec *fsas) {
+  K2_CHECK_EQ(fsas->NumAxes(), 3);
+  ContextPtr c = fsas->Context();
+  int32_t num_fsas = fsas->Dim0(),
+        num_states = fsas->TotSize(1);
+
+  Array1<int32_t> changed(c, 1, 0);
+  Renumbering renumber_states(c, num_states);
+  renumber_states.Keep() = (char)1;  // by default keep all states..
```

Comment:
why not just set 1 in below lambda? I mean, with `num_states` as the num_items of lambda.

Reply:
mm, might be faster for small data but slower for big.. this is not being used right now and would never be a bottleneck, I don't want to spend time optimizing it...